example/threshold_demo.cpp

```diff
@@ -0,0 +1,12 @@
+#include <boost/gil.hpp>
```

Comment:
Please, remove `_demo` suffix from the name. The program is already in `example` folder, so the `_demo` suffix is superfluous.